src/prefect/experimental/cli/worker.py

```diff
@@ -43,13 +46,30 @@ async def start(
     ),
 ):
     try:
-        # TODO: Add ability to discover worker type from existing pool
+        if worker_type is None:
+            async with get_client() as client:
+                work_pool = await client.read_work_pool(work_pool_name=work_pool_name)
+            worker_type = work_pool.type
+            app.console.print(
+                f"Discovered worker type {worker_type!r} for work pool"
+                f" {work_pool.name!r}."
+            )
         Worker = lookup_type(BaseWorker, worker_type)
```

Comment:
I was very confused about `Worker` being a variable and not a statically defined type; can you call it `worker_cls` instead?